src/functional_pipeline/functional_pipeline.m

```diff
@@ -62,20 +65,63 @@
 
 
 % PARSE INPUT
-[subjectDir, varargin{:}] = convertStringsToChars(subjectDir, varargin{:});
+[subjectSelection, varargin{:}] = convertStringsToChars(subjectSelection, varargin{:});
 
-assert(ischar(subjectDir), ...
+assert(ischar(subjectSelection), ...
     'CATO:functional_pipeline:subjectDirNotText', ...
     'subjectDir must be a row vector of characters or string scalar.');
-assert(isdir(subjectDir), ...
+assert(isdir(subjectSelection) | isfile(subjectSelection), ...
     'CATO:functional_pipeline:subjectDirNotDir', ...
-    'subjectDir (%s) is not a directory.', subjectDir);
+    'subjectDir (%s) is not a directory.', subjectSelection);
 
-[configFile, runType, configParamsCl] = parseVarargin(varargin{:});
-% parseVarargin does also error handling.
+%% Group mode
+
+if isfile(subjectSelection)
+    subjectListFile = subjectSelection;
+    fprintf('\n--------strarting group run----------\n')
+
+    % read file
+    fi = fopen(subjectListFile);
+
+    fprintf('Processing subjects in %s\n\n', subjectListFile);
+
+    % read line by line, keep track of processed/failed/unrecognized
+    subjectDir = fgetl(fi);
+    processedSubjects = 0;
+    failedSubjects = 0;
+    unrecognizedSubjects = 0;
+
+    while ischar(subjectDir)
+        if isdir(subjectDir)
+            try
+                configParams = functional_pipeline(subjectDir, varargin{:});
+                processedSubjects = processedSubjects + 1;
+            catch
+                failedSubjects = failedSubjects + 1;            
+            end
+        else
+            unrecognizedSubjects = unrecognizedSubjects + 1;
+            warning('CATO:functional_pipeline:subjectDirNotDir', ...
+                'subjectDir (%s) is not a directory', subjectDir);
+        end
+        subjectDir = fgetl(fi);
+    end
+
+    fprintf('\n---------group run finished----------\n')
+    fprintf('Processed %d subjects, %d failed, %d unrecognized\n\n', ...
```

Comment:
Nice that the code keeps track of the unsuccessful subjects! It might also be a good idea to provide a list of the subject names that were unsuccessful as this might help users to identify issues quickly. Maybe this list can be displayed alongside the statistics?

To keep this CATO usable with this feature, it might be helpful to truncate the list to a manageable size (e.g. 100 or 1,000) when the number of subjects is very large.